### PR TITLE
Source dir example

### LINKED ISSUE #2847: Support source files for external functions - example
We should as noted in  #2145 add an example demonstrating source files for external functions.

### PR DESCRIPTION
**Pull request by HansOlsson**
Closes #2847

### CODE REVIEW

**Review by eshmoylova — APPROVED** [review]
Looks good to me.

**Review by beutlich — COMMENTED** [review]
Just a small remark on consistency: Maybe you could rename Func[12].c to ExternalFunc[12].c to match the other file name ExternalFunc3.c.

File: chapters/functions.tex [hunk 1]
```diff
@@ -2144,7 +2144,10 @@ \subsection{Annotations for External Libraries and Include Files}\label{annotati
   external "C"
     y = ExternalFunc1_ext(x)
       annotation(Library = "ExternalLib1",
-                 Include = "#include \"ExternalFunc1.h\"");
+                 Include = "#include \"ExternalFunc1.h\"",
+                 SourceDirectory =
+                  "modelica://ExternalFunctions/Resources/Source");
+      // The SourceDirectory is the default and thus redunant.
```
Comment (maltelenz) [inline chapters/functions.tex:2150]: Typo, should be "redundant"

**Comment by HansOlsson** [comment]
> Just a small remark on consistency: Maybe you could rename Func[12].c to ExternalFunc[12].c to match the other file name ExternalFunc3.c.

Actually, _that_ inconsistency is by choice.
I thought that since the different C-files are treated differently (including vs. external compilation) it made sense to name them differently.

File: chapters/functions.tex (hunk 1 above)
Comment (HansOlsson) [reply chapters/functions.tex]: Thanks.

**Pull request merged**